### PR TITLE
Checkbox for line numbers hide/show

### PR DESCRIPTION
**Pull request by cryham**
## Proposed changes
I added a checkbox in Settings on Detailed - Diff viewer (DiffViewerSettingsPage) for line numbers hide/show.
Works right for Diff page, but on File tree page it works on next run..

## Screenshot
screen

## Test methodology
Manually.

## Test environment(s)
- GIT 2.21
- Windows 7 SP1

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Editor/FileViewerInternal.cs [hunk 1]
```diff
@@ -112,6 +112,13 @@ public void SetText(string text, Action openWithDifftool, bool isDiff = false)
             _currentViewPositionCache.Capture();
 
             OpenWithDifftool = openWithDifftool;
+            if (!AppSettings.ShowLineNumbers)
+            {
+                isDiff = false;
+            }
+
+            TextEditor.ShowLineNumbers = false;
```
Comment (gerhardol) [inline GitUI/Editor/FileViewerInternal.cs]: As intended?

Comment (cryham) [reply GitUI/Editor/FileViewerInternal.cs]: Yes, this hides an extra 3rd row (after the 2 own with diff) with numbers going 1,2,3..

**Comment by RussKie** [comment]
Could you please describe a use case where a user would wish to disable line numbers in the diff control?
I can understand a desire to hide line number for authors in the Blame control (#6430), but don't see a compelling value here...
Thank you

**Comment by drewnoakes** [comment]
@RussKie one way of looking at this is that it matches the output of `git diff` on the command line. The line numbers are mentioned in the header.

On the other hand it creates yet another option in the already busy options pages.

**Comment by RussKie** [comment]
> one way of looking at this is that it matches the output of `git diff` on the command line. The line numbers are mentioned in the header.

But why?

**Comment by cryham** [comment]
Well, I personally don't need to see line numbers at all, anywhere, mostly. They take up some valuable space (width). Without them I can see more code, e.g. if lines are long, and with file paths on left. But even if all fits, the view is clearer for me without those numbers.
I do occasionally jump to a line by number, but I don't feel like seeing them all the time. I can easily show external diff for better orientation if needed.

**Comment by mstv** [comment]
I suggest to add this setting to the FileDiff's toolbar:
grafik

**Comment by cryham** [comment]
Right, I also think it's a good idea.
I there an icon for line numbers?
Still to do here is making checkbox toggle update the FileViewer (not only Internal one). Currently works after program restart.

**Comment by mstv** [comment]
We prefer to use the [Fugue Icons](https://p.yusukekamiyamane.com/) if applicable.

**Comment by mstv** [comment]
1. Switching line numbers off and on using the toolbar icon does not affect all necessary flags.
2. There's an additional margin when the Commit dialog is opened with line numbers off.
3. The toolbar icon should swap places with the `Show entire file` button.
4. The hash icon has a different meaning in the domain of Git. I would use one showing an enumerated list.

**Comment by RussKie** [comment]
Closing as stale. None of the comments have been addressed.
If you wish to pursue it further please ping us to re-open. 
Thanks

**Pull request closed without merging**